### PR TITLE
Spack images

### PR DESCRIPTION
**Pull request by luca-heltai**
Here's two Dockerfiles to build images based on spack.

### CODE REVIEW

File: full-deps/fulldepsspack/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+              git \
+              build-essential \
+	      gfortran \
+              python \
+              nano \
+              unzip \
+           && rm -rf /var/lib/apt/lists/*
+
+# install spack
+RUN        curl -s -L https://api.github.com/repos/llnl/spack/tarball \
```
Comment (jppelteret) [inline full-deps/fulldepsspack/Dockerfile]: I think that the "ownership" of spack has moved from `llnl` to `spack`, so this should be `https://api.github.com/repos/spack/spack/tarball`. @davydden Do you agree with this?

Comment (davydden) [reply full-deps/fulldepsspack/Dockerfile]: yes, absolutely. It's official Github is https://github.com/spack/spack , that happened like a year ago.

File: full-deps/fulldepsspack/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+# general environment for docker
+ENV        DEBIAN_FRONTEND=noninteractive \
+           SPACK_ROOT=/usr/local \
+           FORCE_UNSAFE_CONFIGURE=1
+
+# install minimal spack dependencies
+RUN        apt-get update \
+           && apt-get install -y --no-install-recommends \
+              autoconf \
```
Comment (davydden) [inline full-deps/fulldepsspack/Dockerfile]: i think some of those are not needed really, `curl` and `git` `gfortran` have to be here, but with the current setup of `packages.yaml` the rest will be built by Spack. One could of course set a couple of those packages as externally provided, but for those small packages it does not take much time to build so i would not bother with this.

File: dealii/fulldepsspack/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+FROM dealii/full-deps:fulldepsspack
+
+MAINTAINER luca.heltai@gmail.com
+
+# get deal.II repo
+USER       root
+ARG 	   VER=develop
+RUN        spack install -j12 dealii@$VER \
```
[the remark this answers is not in the document]
Comment (luca-heltai) [reply dealii/fulldepsspack/Dockerfile]: not under docker...

[the remark this answers is not in the document]
Comment (davydden) [reply dealii/fulldepsspack/Dockerfile]: ah, interesting

File: full-deps/fulldepsspack/Dockerfile (hunk 2 above)
Comment (luca-heltai) [reply full-deps/fulldepsspack/Dockerfile]: which ones do you suggest I remove?

Comment (luca-heltai) [reply full-deps/fulldepsspack/Dockerfile]: all except curl, git and gfortran?

Comment (davydden) [reply full-deps/fulldepsspack/Dockerfile]: i would try without: `autoconf`, `python`, `nano`, `environment-modules`.  But we can merge it as-is and if this list can be shorten, then we can do this in another PR.

Comment (luca-heltai) [reply full-deps/fulldepsspack/Dockerfile]: I still need `build-essentials` and python, right?

File: dealii/fulldepsspack/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (luca-heltai) [reply dealii/fulldepsspack/Dockerfile]: docker thinks it has only 2 cores. If I say -j12 it uses all 12 cores

File: full-deps/fulldepsspack/Dockerfile (hunk 2 above)
Comment (davydden) [reply full-deps/fulldepsspack/Dockerfile]: oh, `python` of course.

Comment (davydden) [reply full-deps/fulldepsspack/Dockerfile]: `build-essentials` as well, i think. So it's only `autoconf`, `nano`, `environment-modules`

**Comment by luca-heltai** [comment]
It turns out I was wrong. docker by default uses all cpus it can. I don't know why, though, spack is still not using them on my machines, unless I force it with -j12.

**Comment by davydden** [comment]
@jppelteret do you have any further comments to the docker image PR? If not, we can merge it. 

@luca-heltai this fixes https://github.com/dealii/dealii/issues/6247 , right?

**Pull request merged**